firebase-firestore/src/test/java/com/google/firebase/firestore/model/MutationSquashTest.java

```diff
@@ -0,0 +1,250 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.model;
+
+import static com.google.firebase.firestore.testutil.TestUtil.deletedDoc;
+import static com.google.firebase.firestore.testutil.TestUtil.doc;
+import static com.google.firebase.firestore.testutil.TestUtil.field;
+import static com.google.firebase.firestore.testutil.TestUtil.key;
+import static com.google.firebase.firestore.testutil.TestUtil.map;
+import static com.google.firebase.firestore.testutil.TestUtil.mergeMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.patchMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.setMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.unknownDoc;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.FieldValue;
+import com.google.firebase.firestore.model.mutation.DeleteMutation;
+import com.google.firebase.firestore.model.mutation.Mutation;
+import com.google.firebase.firestore.model.mutation.MutationSquash;
+import com.google.firebase.firestore.model.mutation.Precondition;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+/** Tests Mutations */
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class MutationSquashTest {
+  @Test
+  public void testSquashOneSetMutation() {
+    Map<String, Object> data = map("foo", "foo-value", "baz", "baz-value");
+    MutableDocument setDoc = doc("collection/key", 0, data);
+    MutableDocument original = setDoc.clone();
+
+    Timestamp now = Timestamp.now();
+    Mutation set = setMutation("collection/key", map("bar", "bar-value"));
+    set.applyToLocalView(setDoc, now, MutationSquash.Type.None);
+
+    Mutation squashed = MutationSquash.toMutation(original, setDoc, MutationSquash.Type.Set);
+    squashed.applyToLocalView(original, now, MutationSquash.Type.None);
+    assertEquals(original, setDoc);
```

Comment:
Can you add a helper here? FWIW this looks very much like `verifyRoundtrip` in our serializer test.